src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -94,96 +114,169 @@ pub enum FilesystemEventKind {
 // ==================
 
 /// Attributes of the file in the filesystem.
-#[derive(Clone,Copy,Debug,PartialEq)]
+#[derive(Clone,Debug,PartialEq,Eq,Hash)]
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Attributes{
     /// When the file was created.
-    pub creation_time      : UTCDateTime,
+    pub creation_time : UTCDateTime,
     /// When the file was last accessed.
-    pub last_access_time   : UTCDateTime,
+    pub last_access_time : UTCDateTime,
     /// When the file was last modified.
     pub last_modified_time : UTCDateTime,
     /// What kind of file is this.
-    pub file_kind          : FileKind,
+    pub kind : FileSystemObject,
     /// Size of the file in bytes.
     /// (size of files not being `RegularFile`s is unspecified).
-    pub byte_size          : u64
+    pub byte_size : u64
 }
 
-/// What kind of file (regular, directory, symlink) is this.
-#[derive(Clone,Copy,Debug,PartialEq)]
-#[derive(Serialize, Deserialize)]
-pub enum FileKind {
-    /// File being a directory.
-    Directory,
-    /// File being a symbolic link.
-    SymbolicLink,
-    /// File being a regular file with opaque content.
-    RegularFile,
-    /// File being none of the above, e.g. a physical device or a pipe.
-    Other
+/// A representation of what kind of type a filesystem object can be.
+#[derive(Hash,Debug,Clone,PartialEq,Eq,Serialize,Deserialize)]
+#[serde(tag = "type")]
+pub enum FileSystemObject {
+    /// Represents a directory.
+    Directory(Object),
+    /// Represents a directory which contents have been truncated
+    DirectoryTruncated(Object),
+    /// Represents a file.
+    File(Object),
+    /// Represents unrecognized object. Example is a broken symbolic link.
+    Other(Object),
```

Comment:
Then it should be named `Unrecognized` instead.

Reply:
@farmaazon Nope, I'm asking those who are doing the code here. If this is connected with engine, the person doing the PAI should talk with them, not the reviewer.

Reply:
> If this is connected with engine, the person doing the PAI should talk with them.
But nobody "is doing API" here, at least not the engine API. We're creating a client to connect with this API. Fixing bad names in engine API is not a part of this task. 

Here we want to keep name consistent with the API, and that is the first priority.

Reply:
I'm only telling this code has bad naming. Imo Danilo should chat with engine about changing the naming or create ticket for them about that in this case + write FIXME in code pointing to the ticket. Anyway, this is not task of the reviewer to involve in all codebase and discussions. Agreeing upon API and naming is between you and engine.

Reply:
So, @notdanilo create a issue for discussing naming of protocol with engine.

Reply:
@farmaazon is this issue created by @mwu-tow enough? https://github.com/luna/enso/issues/708